Exercises/NoModules/Chapter 13/Soln13_03/Box.cpp

```diff
@@ -0,0 +1,77 @@
+#include "Box.h"
+
+#include <format>
+#include <algorithm>    // For the min() and max() function templates
+
+double Box::volume() const
+{
+    return m_length * m_width * m_height;
+}
+
+Box Box::operator+(const Box& aBox) const
+{
+  Box copy{ *this };  // Implement in terms of += operator (avoid duplication)
+  copy += aBox;
+  return copy;        // Return new object (convention)
+}
+
+Box Box::operator*(double factor) const
+{
+  Box copy{ *this };  // Implement in terms of *= operator (avoid duplication)
+  copy *= factor;
+  return copy;        // Return new object (convention)
+}
+
+Box Box::operator/(double divisor) const
+{
+  Box copy{ *this };  // Implement in terms of /= operator (avoid duplication)
+  copy /= divisor;
+  return copy;        // Return new object (convention)
+}
+
+Box& Box::operator+=(const Box& aBox)
+{
+  // New object has larger length and width, and sum of heights
+  m_length = std::max(m_length, aBox.m_length);
+  m_width  = std::max(m_width, aBox.m_width);
+  m_height += aBox.m_height;
+  return *this;   // Return a reference to the left operand (convention)
+}
+
+Box& Box::operator*=(double factor)
+{
+  m_length *= factor;
+  m_width  *= factor;
+  m_height *= factor;
+  return *this;   // Return a reference to the left operand (convention)
+}
+
+Box& Box::operator/=(double divisor)
+{
+  m_length /= divisor;
```

Comment:
check for division by zero?

Reply:
Same as before: exceptions are not known yet. And besides, I prefer this to behave as it does now, precisely like the built-in operator.